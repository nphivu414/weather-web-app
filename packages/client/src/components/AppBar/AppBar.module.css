.appBar {
  min-height: 35px;
}
