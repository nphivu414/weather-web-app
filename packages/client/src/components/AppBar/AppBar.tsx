import * as React from 'react';
import { BiChevronLeft } from 'react-icons/bi';
import styles from './AppBar.module.css';

export type AppBarProps = {
  title: string;
  hasBackButton?: boolean;
  onBack?: () => void;
};

export const handleOnBackButtonClick = (onBack?: () => void) => {
  return () => {
    onBack && onBack();
  };
};

export const AppBar: React.FC<AppBarProps> = ({ title, hasBackButton, onBack }) => {
  return (
    <div className="container bg-primary p-2">
      <div className={[styles.appBar, 'row align-items-center'].join(' ')}>
        <div className="col-3">
          {hasBackButton && (
            <button type="button" className="btn btn-link btn-sm" onClick={handleOnBackButtonClick(onBack)}>
              <BiChevronLeft size="20px" className="align-bottom text-white" />
              <span className="text-white">Back</span>
            </button>
          )}
        </div>
        <div className="col-6 text-center">
          <p className="text-white fw-bold text-truncate m-0">{title}</p>
        </div>
        <div className="col-3" />
      </div>
    </div>
  );
};

export default AppBar;
