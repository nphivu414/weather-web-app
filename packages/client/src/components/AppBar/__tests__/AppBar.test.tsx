import * as React from 'react';
import { shallow } from 'enzyme';

import AppBar, { AppBarProps } from '../AppBar';

const props: AppBarProps = {
  title: 'Home',
};

describe('AppBar', () => {
  it('should match a snapshot', () => {
    expect(shallow(<AppBar {...props} />)).toMatchSnapshot();
  });
  it('should show back button', () => {
    const wrapper = shallow(<AppBar {...props} hasBackButton />);
    expect(wrapper.find('button.btn')).toHaveLength(1);
  });
});
