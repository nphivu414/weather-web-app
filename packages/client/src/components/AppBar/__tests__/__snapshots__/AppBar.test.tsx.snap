// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should match a snapshot 1`] = `ShallowWrapper {}`;
